src/BoilerPlate/Asteroid.cpp

```diff
@@ -1,23 +1,38 @@
 #include "Asteroid.hpp"
-#include "Vector2.hpp"
+
 #include "MathUtilities.hpp"
+#include "Vector2.hpp"
+
 #include <GL/glew.h>
 #include <SDL2/SDL_opengl.h>
 
 #include <vector>
 #include <cmath>
-#include <ctime>
-#include <cstdio>
+
+const int MAX_SPEED = 500;
 
 std::vector <Vector2> asteroidPoints;
 
+inline float getSizeFactor(AsteroidSize size) {
+	return 2.0f*static_cast<int>(size) + 1;
+}
 
-Asteroid::Asteroid (AsteroidSize _size, int width, int height)
-	: SpaceObject(width,height), m_asteroidSize(_size)
+Asteroid::Asteroid (AsteroidSize size, int width, int height)
```

Comment:
Are width and height relevant to the size of the asteroid or to the viewport? If it's the latter then please add a pointer to the entity so it knows about its parent class which should be Game, and then query the screen size from it.